src/modules/gateway/sagas.ts

```diff
@@ -83,268 +100,341 @@ export function createGatewaySaga(config: ManaFiatGatewaySagasConfig) {
       yield put(setPurchase(purchase))
     }
   }
-}
-
-function* handleOpenBuyManaWithFiatModal(
-  config: ManaFiatGatewaySagasConfig,
-  action: OpenBuyManaWithFiatModalRequestAction
-) {
-  try {
-    const { selectedNetwork } = action.payload
-    const pendingManaPurchase: Purchase | undefined = yield select(
-      getPendingManaPurchase
-    )
 
-    if (pendingManaPurchase) {
-      let goToUrl: string | undefined
+  function* handleOpenFiatGatewayWidget(
+    action: OpenFiatGatewayWidgetRequestAction
+  ) {
+    const { data, gateway, listeners } = action.payload
+    try {
+      switch (gateway) {
+        case FiatGateway.WERT:
+          const { onLoaded, onPending, onSuccess } = listeners || {}
+          const { marketplaceServerURL } = config[FiatGateway.WERT]
+
+          const wallet: Wallet | null = yield select(getWalletData)
+          if (wallet) {
+            const identity: AuthIdentity | null = yield call(
+              getIdentityOrRedirect
+            )
 
-      if (pendingManaPurchase.gateway === NetworkGatewayType.MOON_PAY) {
-        goToUrl = new MoonPay(config.moonPay).getTransactionReceiptUrl(
-          pendingManaPurchase.id
-        )
+            if (!identity) {
+              yield put(openFiatGatewayWidgetFailure(NO_IDENTITY_ERROR))
+              return
+            } else if (identity) {
```

Comment:
WDYT about simplifying this?:
```suggestion
            if (!identity) {
              yield put(openFiatGatewayWidgetFailure(NO_IDENTITY_ERROR))
              return
            }
```